### PR TITLE
Fix 01114_database_atomic flakiness

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix 01114_database_atomic flakiness

https://s3.amazonaws.com/clickhouse-test-reports/67983/e8d9fbdd1f9bd23ee9fda60e2068b4a9642e52cd/stateless_tests__asan__[2_2].html

Sleep is a terrible synchronization method

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 2d2e8b3a967b5455870d6a9dd873c1deeaf77af3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68930/2d2e8b3a967b5455870d6a9dd873c1deeaf77af3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/docs_check.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Integration tests</td><td>The integration tests report. In parenthesis the package type is given, and in square brackets are the optional part/total tests</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/integration_tests__aarch64__[1_6].html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/6289c65e0286127689303b5b7a543212ca38e0c7/unit_tests__asan_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by Algunenano** [comment]
[From failed flaky check](https://s3.amazonaws.com/clickhouse-test-reports/68930/2d2e8b3a967b5455870d6a9dd873c1deeaf77af3/stateless_tests_flaky_check__asan_.html) there is something unrelated to the test changes but quite intriguing. Why did this query take 165 seconds instead of the expected 33?

```
$ grep 23890d4e-3387-4c3d-8cca-a79984428744 a.log
2024.08.27 07:42:13.955871 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> executeQuery: (from [::1]:42968) (comment: 01114_database_atomic.sh) SELECT count(col), sum(col) FROM (SELECT n + sleepEachRow(1.5) AS col FROM test_cb36bnb2_1.mt) -- 25279 (stage: Complete)
2024.08.27 07:42:13.977383 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Planner: Query to stage Complete
2024.08.27 07:42:13.979284 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Planner: Query to stage Complete
2024.08.27 07:42:13.988626 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Planner: Query from stage FetchColumns to stage Complete
2024.08.27 07:42:13.991501 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Planner: Query from stage FetchColumns to stage Complete
2024.08.27 07:42:14.008912 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> test_cb36bnb2_3.mt (SelectExecutor): Key condition: unknown
2024.08.27 07:42:14.015499 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> test_cb36bnb2_3.mt (SelectExecutor): MinMax index condition: unknown
2024.08.27 07:42:14.017124 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Filtering marks by primary and secondary keys
2024.08.27 07:42:14.020040 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> test_cb36bnb2_3.mt (SelectExecutor): Selected 5/5 parts by partition key, 5 parts by primary key, 5/5 marks by primary key, 5 marks to read from 5 ranges
2024.08.27 07:42:14.023495 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Spreading mark ranges among streams (default reading)
2024.08.27 07:42:14.026671 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Reading 1 ranges in order from part 0_1_1_0, approx. 22 rows starting from 0
2024.08.27 07:42:14.028694 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Reading 1 ranges in order from part 1_2_2_0, approx. 22 rows starting from 0
2024.08.27 07:42:14.031261 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Reading 1 ranges in order from part 2_3_3_0, approx. 22 rows starting from 0
2024.08.27 07:42:14.033017 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Reading 1 ranges in order from part 3_4_4_0, approx. 22 rows starting from 0
2024.08.27 07:42:14.041174 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> test_cb36bnb2_3.mt (SelectExecutor): Reading 1 ranges in order from part 4_5_5_0, approx. 22 rows starting from 0
2024.08.27 07:42:47.069202 [ 1435 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> AggregatingTransform: Aggregating
2024.08.27 07:42:47.070030 [ 1435 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Aggregator: Aggregation method: without_key
2024.08.27 07:43:56.968718 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> TCPHandler: Sending profile events block with 5 rows, 374 bytes took 137 milliseconds
2024.08.27 07:44:18.128161 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> TCPHandler: Sending profile events block with 5 rows, 374 bytes took 105 milliseconds
2024.08.27 07:44:59.126268 [ 1435 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> AggregatingTransform: Aggregated. 110 to 1 rows (from 880.00 B) in 165.079057233 sec. (0.666 rows/sec., 5.33 B/sec.)
2024.08.27 07:44:59.127164 [ 1435 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> Aggregator: Merging aggregated data
2024.08.27 07:44:59.128130 [ 1435 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Trace> HashTablesStatistics: Statistics updated for key=8495912807294640416: new sum_of_sizes=1, median_size=1
2024.08.27 07:44:59.146471 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> executeQuery: Read 110 rows, 880.00 B in 165.222086 sec., 0.6657705556386693 rows/sec., 5.33 B/sec.
2024.08.27 07:44:59.151001 [ 2370 ] {23890d4e-3387-4c3d-8cca-a79984428744} <Debug> TCPHandler: Processed in 165.301555906 sec
```

Well, 33 is when all partitions are processed in parallel, not when they are all executed in the same thread. And in these cases we have `--max_threads 1`. I'll adjust it so we use more threads on those queries. It might still be possible to be slower than 33 seconds, which is ok, but it won't take 3 minutes to run the test.

File: tests/queries/0_stateless/01114_database_atomic.sh [hunk 1]
```diff
@@ -9,81 +9,98 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
 # shellcheck source=../shell_config.sh
 . "$CURDIR"/../shell_config.sh
 
+DATABASE_1="${CLICKHOUSE_DATABASE}_1"
+DATABASE_2="${CLICKHOUSE_DATABASE}_2"
+DATABASE_3="${CLICKHOUSE_DATABASE}_3"
```
Comment (Avogar) [inline tests/queries/0_stateless/01114_database_atomic.sh:14]: As I see in the reference file, `CLICKHOUSE_DATABASE` is actually just `default`. It's a bit strange, I would expect it to be `test_{random_str}`. With databases `default_1/2/3` I don't see much difference from `test_01114_1/2/3`

File: tests/queries/0_stateless/01114_database_atomic.sh [hunk 2]
```diff
@@ -9,81 +9,98 @@ CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
...
 
-uuid_db_1=`$CLICKHOUSE_CLIENT -q "SELECT uuid FROM system.databases WHERE name='test_01114_1'"`
-uuid_db_2=`$CLICKHOUSE_CLIENT -q "SELECT uuid FROM system.databases WHERE name='test_01114_2'"`
+uuid_db_1=`$CLICKHOUSE_CLIENT -q "SELECT uuid FROM system.databases WHERE name='${DATABASE_1}'"`
+uuid_db_2=`$CLICKHOUSE_CLIENT -q "SELECT uuid FROM system.databases WHERE name='${DATABASE_2}'"`
 $CLICKHOUSE_CLIENT -q "SELECT name,
                               engine,
                               splitByChar('/', data_path)[-2],
                               splitByChar('/', metadata_path)[-2] as uuid_path, ((splitByChar('/', metadata_path)[-3] as metadata) = substr(uuid_path, 1, 3)) OR metadata='metadata'
                               FROM system.databases WHERE name LIKE 'test_01114_%'" | sed "s/$uuid_db_1/00001114-1000-4000-8000-000000000001/g" | sed "s/$uuid_db_2/00001114-1000-4000-8000-000000000002/g"
```
Comment (Avogar) [inline tests/queries/0_stateless/01114_database_atomic.sh:32]: Should we change `test_01114` here to `CLICKHOUSE_DATABASE`?

File: tests/queries/0_stateless/01114_database_atomic.sh (hunk 1 above)
Comment (Avogar) [reply tests/queries/0_stateless/01114_database_atomic.sh:14]: Ah, we replace `test_{random_str}` to `default` in the test results. Did't know that

File: tests/queries/0_stateless/01114_database_atomic.sh (hunk 2 above)
Comment (Algunenano) [reply tests/queries/0_stateless/01114_database_atomic.sh:32]: Yes, I missed it. Thanks!

**Review by Avogar — APPROVED** [review]

**Comment by Algunenano** [comment]
Failure in `01290_max_execution_speed_distributed` unrelated. I'll fix in in a different PR

**Pull request merged**